src/app/public/src/modules/shared/widget.service.ts

```diff
@@ -13,6 +15,21 @@ export class HelpWidgetService {
     return this._pageDefaultKey;
   }
 
+  public increaseDisabledCount(): void {
```

Comment:
I'm fine with whichever strategy you use to figure out when the widget should be disabled (in this case, a counter), but I feel strongly that this implementation detail shouldn't be exposed to the consumers of this service. For example, other consumers of the service might not know when to call `disableWidget` or `increaseDisabledCount`. Would it be possible to add the counter stuff to the current `disableWidget` method so that all the consumers of the service would need to do is call disable or enable?

```
public disableWidget() {
  this.disabledCount++;
  return this.executeWhenClientReady(() => {
  // ...
}

public enableWidget() {
  if (this.disabledCount > 0) {
    this.disabledCount--;
  }

  if (this.disabledCount === 0) {
    return this.executeWhenClientReady(() => {
    // ...
  }
}
```